qlib/rl/order_execution/simulator_qlib.py

```diff
@@ -2,3 +2,199 @@
 # Licensed under the MIT License.
 
 """Placeholder for qlib-based simulator."""
+import copy
+from typing import Callable, Generator, List, Optional, Tuple, Union
+
+import pandas as pd
+from gym.vector.utils import spaces
+
+from qlib.backtest import get_exchange
+from qlib.backtest.account import Account
+from qlib.backtest.decision import Order, TradeRange, TradeRangeByTime
+from qlib.backtest.executor import BaseExecutor
+from qlib.backtest.utils import CommonInfrastructure
+from qlib.config import QlibConfig
+from qlib.rl.interpreter import ActionInterpreter
+from qlib.rl.order_execution.from_neutrader.config import ExchangeConfig
+from qlib.rl.order_execution.from_neutrader.executor import RLNestedExecutor
+from qlib.rl.order_execution.from_neutrader.feature import init_qlib
+from qlib.rl.order_execution.from_neutrader.state import SAOEEpisodicState
+from qlib.rl.order_execution.from_neutrader.strategy import DecomposedStrategy
+from qlib.rl.simulator import Simulator
+from qlib.strategy.base import BaseStrategy
+
+
+def get_common_infra(
+    config: ExchangeConfig,
+    trade_start_time: pd.Timestamp,
+    trade_end_time: pd.Timestamp,
+    codes: List[str],
+    cash_limit: Optional[float] = None,
+) -> CommonInfrastructure:
+    # need to specify a range here for acceleration
+    if cash_limit is None:
+        trade_account = Account(
+            init_cash=int(1e12),
+            benchmark_config={},
+            pos_type='InfPosition'
+        )
+    else:
+        trade_account = Account(
+            init_cash=cash_limit,
+            benchmark_config={},
+            pos_type='Position',
+            position_dict={code: {"amount": 1e12, "price": 1.} for code in codes}
+        )
+
+    exchange = get_exchange(
+        codes=codes,
+        freq='1min',
+        limit_threshold=config.limit_threshold,
+        deal_price=config.deal_price,
+        open_cost=config.open_cost,
+        close_cost=config.close_cost,
+        min_cost=config.min_cost if config.trade_unit is not None else 0,
+        start_time=pd.Timestamp(trade_start_time),
+        end_time=pd.Timestamp(trade_end_time) + pd.DateOffset(1),
+        trade_unit=config.trade_unit,
+        volume_threshold=config.volume_threshold
+    )
+
+    return CommonInfrastructure(trade_account=trade_account, trade_exchange=exchange)
+
+
+class CategoricalActionInterpreter(ActionInterpreter[SAOEEpisodicState, int, float]):
```

Comment:
If `SAOEState` is used, you can directly use interpreters from `interpreter.py` here.